lucene/analysis/common/src/java/org/apache/lucene/analysis/hunspell/Stemmer.java

```diff
@@ -319,6 +345,11 @@ private boolean containsSharpS(char[] word, int offset, int length) {
     return deduped;
   }
 
+  interface RootProcessor {
+    /** @return whether the processing should be continued */
+    boolean processRoot(char[] buffer, int offset, int length, IntsRef forms, int formID);
```

Comment:
Use CharsRef instead of three separate parameters here? Shortens parameters list a bit. On the calling side it can be a short-lived wrapper (new CharsRef(...)).

Reply:
Indeed the performance effects appear to be negligible, thanks!